chemberta/utils/raw_text_dataset.py

```diff
@@ -41,9 +44,9 @@ def __init__(self, tokenizer, file_path: str, block_size: int):
     def __len__(self):
         return self.len
 
-    def preprocess(self, text):
+    def preprocess(self, feature_dict):
         batch_encoding = self.tokenizer(
-            str(text),
+            feature_dict["text"],
```

Comment:
I discovered that the arg to `preprocess` is actually a feature dictionary. Previously, the dataset code was coercing this to a string representation that contained extraneous characters; e.g.:

```
>> print(str(text))
"{'text': 'CCC(CC)(Pc1ccc(C)cc1CO)c1cc(OC)ccc1OCOC'}"
```

Reply:
very good catch